### PR TITLE
Add an integration test case for user account locking

### PR DESCRIPTION
**Pull request by ShanChathusanda93**
## Changes
Added an integration test to check whether a user account is locked while the caseInsensitiveUsername and useCaseSensitiveUsernameCacheKey properties are false in user store.
Git issue : https://github.com/wso2/product-is/issues/7169

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/identity/mgt/AccountLockingWhileCaseInsensitiveUserFalseTestCase.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+import org.wso2.identity.integration.common.utils.ISIntegrationTest;
+import org.wso2.identity.integration.test.util.Utils;
+
+import java.io.File;
+
+/**
+ * This test class is to test the user account locking process while the caseInsensitiveUserName and
+ * useCaseSensitiveUsernameForCacheKey properties are false in the primary user store.
+ */
+public class AccountLockingWhileCaseInsensitiveUserFalseTestCase extends ISIntegrationTest {
```
Comment (DMHP) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/identity/mgt/AccountLockingWhileCaseInsensitiveUserFalseTestCase.java]: Shall we change the class name to AccountLockWhileCaseInsensitiveUserFalseTestCase

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/identity/mgt/AccountLockingWhileCaseInsensitiveUserFalseTestCase.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+    @BeforeClass(alwaysRun = true)
+    public void testInit() throws Exception {
+
+        super.init();
+        String carbonHome = Utils.getResidentCarbonHome();
+
+        File defaultTomlFile = getDeploymentTomlFile(carbonHome);
+        File configuredTomlFile = new File(getISResourceLocation() + File.separator + "identityMgt"
+                + File.separator + "case_insensitive_user_false.toml");
```
Comment (DMHP) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/identity/mgt/AccountLockingWhileCaseInsensitiveUserFalseTestCase.java]: shall we re-use the method getDeploymentTomlFile() defined in ISIntegrationTest base class to fetch the file.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/identity/mgt/AccountLockingWhileCaseInsensitiveUserFalseTestCase.java (hunk 1 above)
Comment (ShanChathusanda93) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/identity/mgt/AccountLockingWhileCaseInsensitiveUserFalseTestCase.java]: Updated

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/identity/mgt/AccountLockingWhileCaseInsensitiveUserFalseTestCase.java (hunk 2 above)
Comment (ShanChathusanda93) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/identity/mgt/AccountLockingWhileCaseInsensitiveUserFalseTestCase.java]: Updated

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/identity/mgt/AccountLockWhileCaseInsensitiveUserFalseTestCase.java [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    private RemoteUserStoreManagerServiceClient usmServiceClient;
+    private AuthenticatorClient authClient;
+
+    @BeforeClass(alwaysRun = true)
+    public void testInit() throws Exception {
+
+        super.init();
+        String carbonHome = Utils.getResidentCarbonHome();
+        configureServerWithRestart(carbonHome);
+
```
Comment (DMHP) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/identity/mgt/AccountLockWhileCaseInsensitiveUserFalseTestCase.java]: Shall we put a comment like "Initiating after the restart."

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/identity/mgt/AccountLockWhileCaseInsensitiveUserFalseTestCase.java [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+        configurationManager = new ServerConfigurationManager(isServer);
+        configurationManager.applyConfigurationWithoutRestart(configuredTomlFile, defaultTomlFile, true);
+        configurationManager.restartGracefully();
+        log.info("Toml configurations applied.");
+    }
+
+    private void createLockedUser(String userName, String[] roles, String password) {
+
+        log.info("Creating a locked user account.");
+        ClaimValue[] claimValues = new ClaimValue[1];
```
Comment (DMHP) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/identity/mgt/AccountLockWhileCaseInsensitiveUserFalseTestCase.java]: Shall we extract [1] to a private method.

[1] https://github.com/wso2/product-is/pull/7331/files#diff-5a568678ef146a2ce34d484b6f8b09fdR106-R110

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/identity/mgt/AccountLockWhileCaseInsensitiveUserFalseTestCase.java [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+        ClaimValue lockedClaim = new ClaimValue();
+        lockedClaim.setClaimURI(ACCOUNT_LOCK_CLAIM_URI);
+        lockedClaim.setValue(Boolean.TRUE.toString());
+        claimValues[0] = lockedClaim;
+
+        try {
+            usmServiceClient.addUser(userName, password, roles, claimValues, null, false);
+            log.info("Locked user account created.");
+        } catch (Exception e) {
```
Comment (DMHP) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/identity/mgt/AccountLockWhileCaseInsensitiveUserFalseTestCase.java]: Shall we catch only the specific exception types.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/identity/mgt/AccountLockWhileCaseInsensitiveUserFalseTestCase.java (hunk 3 above)
Comment (ShanChathusanda93) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/identity/mgt/AccountLockWhileCaseInsensitiveUserFalseTestCase.java]: Updated.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/identity/mgt/AccountLockWhileCaseInsensitiveUserFalseTestCase.java (hunk 4 above)
Comment (ShanChathusanda93) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/identity/mgt/AccountLockWhileCaseInsensitiveUserFalseTestCase.java]: Updated.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/identity/mgt/AccountLockWhileCaseInsensitiveUserFalseTestCase.java (hunk 5 above)
Comment (ShanChathusanda93) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/identity/mgt/AccountLockWhileCaseInsensitiveUserFalseTestCase.java]: Updated.

File: modules/integration/tests-integration/tests-backend/src/test/resources/testng.xml [hunk 6]
```diff
@@ -245,6 +245,7 @@
             <class name="org.wso2.identity.integration.test.user.profile.mgt.UserProfileAdminTestCase"/>
             <!-- TODO this test has a restart-->
             <class name="org.wso2.identity.integration.test.auth.RiskBasedLoginTestCase" />
+            <class name="org.wso2.identity.integration.test.identity.mgt.AccountLockWhileCaseInsensitiveUserFalseTestCase"/>
```
Comment (DMHP) [inline modules/integration/tests-integration/tests-backend/src/test/resources/testng.xml]: As this test case contains a re-start this test should be fall into the 'is-tests-with-individual-configuration-changes' group.

**Pull request merged**